simulation.cpp

```diff
@@ -51,13 +51,13 @@ simulation::simulation(all_params params):
 
 const std::vector<double>& simulation::get_inds_input() const
 {
-  assert(all_individuals_have_same_input(*this));
-  return get_inds()[0].get_input_values();
+    assert(all_individuals_have_same_input(*this));
+    return get_inds()[0].get_input_values();
 }
 
 const std::vector<individual> &simulation::get_inds() const
```

Comment:
🏇

Reply:
Wild and free...

Reply:
I've done the others, and I tried this one quickly but if we want that I think it should be its own issue because it's breaking everything

Reply:
I had a look, I does not seem to break anything on my machine. I also had a look, and the free function version of this function already exists, so I just deleted this one and switched with the free function in the 2 instances it was called. I suspect it was breaking becaus of by turning it into a free function you handed up having to functions with the same name.
Also I found this unusual, in:
In
```c++
const std::vector<individual> &simulation::get_inds() const
{
    return simulation::get_pop().get_inds();
}
```
we call the function `simulation::get_pop()` using the simulation::` namespace to get the population object, which could actually have been called just with
```suggestion
const std::vector<individual> &simulation::get_inds() const
{
    return get_pop().get_inds();
}
```

Reply:
Bottomline/tip: remember to check if the free function already exists, and check where the function is called sometimes a function becomes obsolete without us noticing and we could just delete it!